tests/circuit_gen.py

```diff
@@ -0,0 +1,83 @@
+import os
+import math
+import random
+import time
+
+def build_random_qasm(qubits: int, complexity: int, gateset: str) -> None:
```

Comment:
Maybe simpler to use `num_gates` directly as a parameter instead of complexity. Also, to take in a gate set as a list of gates instead of a name. In the actual testing code, we can certainly use these pre-defined gate set constants but would be nice for this method to be general.

nit: `num_qubits` instead of `qubits`